src/uniqlist.c

```diff
@@ -65,7 +65,7 @@ SEXP uniqlist(SEXP l, SEXP order)
       }
     } break;
     case STRSXP : {
-      SEXP *vd=(SEXP *)DATAPTR(v), prev, elem;   // TODO: tried to replace DATAPTR here but (SEXP *)&STRING_ELT(v,0) results in lvalue required as unary ‘&’ operand
+      SEXP *vd=(SEXP *)CHAR(v), prev, elem;
```

Comment:
`CHAR` I found in http://biostat.mc.vanderbilt.edu/wiki/Main/RInternals please take a look before merging @mattdowle

Reply:
See https://github.com/Rdatatable/data.table/pull/3350#pullrequestreview-199731517